Program.cs

```
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InventoryManagementSystem
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        { 
+            Inventory inventory = new Inventory();
+            int choice = 0;
+
+            Console.WriteLine("Hello! Welcome to your inventory!");
+            Console.WriteLine("Follow to the menu below, please");
+
+            while (true)
+            {
+                Console.WriteLine("\n\nEnter the number of the operation to the console\n");
+                Console.WriteLine("1- Add product");
+                Console.WriteLine("2- Find product");
+                Console.WriteLine("3- Remove product");
+                Console.WriteLine("4- Edit product");
+                Console.WriteLine("5- View all product");
+                Console.WriteLine("6- Exit");
+
+                choice  = Convert.ToInt32(Console.ReadLine()); 
+
+                if (choice == 1)
+                {
+                    Console.WriteLine("Enter name:");
+                    string name = Console.ReadLine();
+
+                    Console.WriteLine("Enter Price:");
+                    var price = Console.ReadLine();
+
+                    Console.WriteLine("Enter Quantity:");
+                    var quantity = Console.ReadLine();
+
+                    if (!name.IsNameValid()) Console.WriteLine("Error: Please enter a valid name.");
```

1. let's utilize Functions to encapsulate such complexity/(use a proper abstraction level and not dive into relatively low level details in the function context. 
2. let's return and fail early to get read of these nested else/if statement. 
3. don't wanna dive into this but let's discuss that the validation could happen on different levels, (Input Validation,  Domain Validation).